### ISSUE TITLE
Reproducibility 

### ISSUE THREAD

**Issue by Johswald** [montrealrobotics/La-MAML#2]
Hey! Thank you very much for the code base!

I am having troubles reproducing the results in the paper / setting the hps for the cifar and imagenet datasets.
Could you provide the calls to do so? Also the hyperparameters mentioned in the appendix do not match the ones in run_experiments.sh.

For example, if I execute (as in your run_experiments.sh file)

python3 main.py $CIFAR --model lamaml_cifar --expt_name lamaml --memories 200 --batch_size 10 --replay_batch_size 10 --n_epochs 1 \
                    --opt_lr 0.25 --alpha_init 0.1 --opt_wt 0.1 --glances 10 --loader class_incremental_loader --increment 5 \
                    --arch "pc_cnn" --cifar_batches 5 --learn_lr --log_every 3125 --second_order --class_order random \
                    --seed $SEED --grad_clip_norm 1.0 --calc_test_accuracy --validation 0.1

I do get something like ~57% which is outside the margin of the reported results. Also you mention that you use gradient clipping at 2, other learning rates etc. Also in your appendix you report to use 10 glances in the multipass whereas in the run_experiments file 1 glance is configured. I get similar non-matching results for Cifar multipass and tinyimagenet. 

It would be great to have run_experiments.sh that gives the hps to reproduce the results in the paper.
Thank you very much!

**Comment by gunshi** [comment]
Hey Johannes,

Thanks for taking the time to reproduce and use our work :)
I verified the following things:

1. I compared the hyperparams of the paper's table and the sh file - it turns out the only hyperparameters that are different are for C-MAML (--opt_wt) and La-MAML (--opt_lr), the the difference is very small (~0.05).  I ran the commands for C-MAML and La-MAML with the hyperparams from the paper as well as from the commands in the sh, and both times I get the same result (since the methods aren't very sensitive to the learning rates and a difference of <0.1 in the learning rate is too small to affect the result). 

2. Clarification about glances=10: the table in the paper indicates that the epochs or glances for an experiment are listed out, _whichever applies in that particular situation_. 
This means that for the multi-pass case, the indicated quantity is the number of epochs (=10) with glances =1 implicitly since multi-pass case already involves seeing the samples many times through having >1 epochs. Similarly for the single-pass setting the value indicates what the number of glances are, since epochs=1 anyway. Therefore the value of that arg is consistent across the table and the paper.

Is it possible that you changed the commands to have glances =10 in the multi-pass case due to a misinterpretation of the "epochs/glances" column of the table?

Please feel free to follow up with further questions and I hope this helps.

Gunshi

**Comment by ykarmesh** [comment]
I believe Gunshi's comment solves the issue and I am therefore closing this!
